core/src/test/scala/integration/kafka/api/AdminClientIntegrationTest.scala

```diff
@@ -698,6 +701,202 @@ class AdminClientIntegrationTest extends KafkaServerTestHarness with Logging {
     }
   }
 
+  val topic = "topic"
+  val partition = 0
+  val topicPartition = new TopicPartition(topic, partition)
+
+  @Test
+  def testSeekToBeginningAfterDeleteRecords(): Unit = {
+    TestUtils.createTopic(zkUtils, topic, 2, serverCount, servers)
+
+    client = AdminClient.create(createConfig)
+
+    val consumer = consumers.head
+    subscribeAndWaitForAssignment(topic, consumer)
+
+    sendRecords(producers.head, 10, topicPartition)
+    consumer.seekToBeginning(Collections.singleton(topicPartition))
+    assertEquals(0L, consumer.position(topicPartition))
+
+    val result = client.deleteRecords(Map(topicPartition -> RecordsToDelete.beforeOffset(5L)).asJava)
+    val lowWatermark = result.lowWatermarks().get(topicPartition).get()
+    assertEquals(5L, lowWatermark)
+    consumer.seekToBeginning(Collections.singletonList(topicPartition))
+    assertEquals(5L, consumer.position(topicPartition))
+
+    client.close()
+  }
+
+  @Test
+  def testConsumeAfterDeleteRecords(): Unit = {
```

Comment:
It's not so clear to me the sequence you are proposing here in relation to what the actual test does. Can you elaborate more please ?

Reply:
The first test checks that after the deletion, lw has updated and hence seek-to-beginning will respect this new lw.

The second test checks that after the deletion, lw has updated and hence seek-to-offset will respect this new lw.

I'm thinking maybe we can consolidate thest two test to 1) call delete records, 2) seek-to-beginning and check position, 3) seek-to-offset and check the position.

BTW I also think we do not need consumer poll checking that the expected num. messages are returned, since "fetching from the position to leo returns these many num. records" is covered by the consumer unit test already, so this sounds a bit overlap to me.

Reply:
Ah Ok. Got it ! just done ;)